reactioncommerce/reaction
# Accounts dashboard base UI for Feature: "Marketplace and Shop owners should be able to create custom permission groups"

Relates to (and partly resolves) #2184 and #2194 

This PR implements new layout from the designs here. It however focuses on the basic/core functionalities for Permission Groups management. More updates and styling needs to be done as marketplace work progresses.

**To Test**
- Clone this branch, and reaction reset
- Login as admin & open Accounts dashboard
- In place of the old look, you should see the new default "Owner" group showing (in a Sortable table). It should show the default admin account under owner table
- Click on the heading of the "Owner" table (i.e the part with text "Owner). 
- The ActionView should slide out. It should show 3 sections.
- The first section is the "Add Admin User" section. Test it out. It should send an invitation successfully to an email filled in. Test with an existing email too. Confirm that the inline Alerts show based on success or error. Note if there are errors in the browser console too.
- NB: When you send an invite, you currently won't see the account show under any group. This will be fixed in the ticket about "inviting a user directly as part of a group"
- The second section is "Edit Group" section. It should show the list of current groups (plus a "New Group" item); and a Permissions list block of switches.
  - The group that was selected to open the actionView should be highlighted bold. It's active permissions should also be reflected in the Permissions block
  - NB: The design goal is to make the Permissions block show directly under each group, as opposed to under the entire group list. This will come in later.
  - Toggling Permissions to on and off should work
  - Clicking on the pencil icon in from of a group name should show a form to edit the name and description (with the current values in place)
  - Clicking New Group should open similar form with blank values, and all permissions toggled off
  - Test out creating new groups. Confirm it works, with no errors
  - With more groups, switching a user from one group to the other (via the group dropdown in the account row) should work. (NB: Currently, you can move the default admin from owner to another group without restriction. Work will be done on this later.
- The third section is to add users to a group (the currently selected group). 
  - If you have admin users, they'll show up in the user list here. (NB: Review how we [decide admin users](https://github.com/reactioncommerce/reaction/pull/2543/files#diff-f122e684bf1129f007c2ac9153ad6d9dR25))
  - Hovering over a user, the group name changes to action to `Make` the user part of the new group. Clicking this should add the user to that group




##### Edited by @impactmass
